test/TestApps/PushNotificationsDemoApp/main.cpp

```diff
@@ -0,0 +1,142 @@
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::Storage::Streams;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel::Background; // BackgroundTask APIs
+
+time_t ltime;
+char buf[SIZE];
+
+winrt::Windows::Foundation::IAsyncOperation<PushNotificationChannel> RequestChannelAsync()
+{
+    // Register the AAD RemoteIdentifier for the App to receive Push
+    auto channelOperation = PushNotificationManager::CreateChannelAsync(
+        winrt::guid("ccd2ae3f-764f-4ae3-be45-9804761b28b2"));
+
+    // Setup the inprogress event handler
+    channelOperation.Progress(
+        [](auto&& sender, auto&& args)
+        {
+            if (args.status == PushNotificationChannelStatus::InProgress)
+            {
+                // This is basically a noop since it isn't really an error state
+                std::cout << "Channel request is in progress." << std::endl << std::endl;
+            }
+            else if (args.status == PushNotificationChannelStatus::InProgressRetry)
+            {
+                LOG_HR_MSG(
+                    args.extendedError,
+                    "The channel request is in back-off retry mode because of a retryable error! Expect delays in acquiring it. RetryCount = %d",
+                    args.retryCount);
+            }
+        });
+
+    auto result = co_await channelOperation;
+
+    if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+    {
+        auto channelUri = result.Channel().Uri();
+
+        std::cout << "channelUri: " << winrt::to_string(channelUri.ToString()) << std::endl << std::endl;
+
+        auto channelExpiry = result.Channel().ExpirationTime();
+
+        // Register Push Event for Foreground
+        result.Channel().PushReceived([](const auto&, PushNotificationReceivedEventArgs const& args)
+            {
+                auto payload = args.Payload();
+
+                // Do stuff to process the raw payload
+                std::string payloadString(payload.begin(), payload.end());
+                std::cout << "Push notification content received from FOREGROUND: " << payloadString << std::endl << std::endl;
+                args.Handled(true);
+            });
+        // Caller's responsibility to keep the channel alive
+        co_return result.Channel();
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with channel request!");
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task = RequestChannelAsync();
+    if (task.wait_for(std::chrono::seconds(960)) != AsyncStatus::Completed)
```

Comment:
Seriously, nobody is going to way 960 seconds for their app to initialize.

Reply:
Changed to 300 seconds.